src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/InterpreterCallInterceptor.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.IL;
+using Internal.Runtime.CallConverter;
+using Internal.Runtime.CallInterceptor;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace Internal.Runtime.Interpreter
+{
+    public class InterpreterCallInterceptor : CallInterceptor.CallInterceptor
+    {
+        private readonly MethodDesc _method;
+        private readonly MethodIL _methodIL;
+        private readonly TypeSystemContext _context;
+
+        public InterpreterCallInterceptor(TypeSystemContext context, MethodDesc method) : base(false)
+        {
+            _context = context;
+            _method = method;
+            _methodIL = EcmaMethodIL.Create(method as EcmaMethod);
+        }
+
+        public override LocalVariableType[] ArgumentAndReturnTypes
+        {
+            get
+            {
+                LocalVariableType[] localVariableTypes = new LocalVariableType[_method.Signature.Length + 1];
```

Comment:
For instance methods, you'll likely need another slot (unless the call interceptor does that for you)